lab_1/main.py

```diff
@@ -40,32 +77,76 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
     :param top_n: a number of the most common words
     :return: a list of the most common words
     """
-    pass
+    # Check for bad input
+    if (not isinstance(freq_dict, dict)
+            or not isinstance(top_n, int)):
+        return None
+    # Convert frequency dictionary to list of tuples
+    tokens = freq_dict.items()
+    # Sort list by frequency
+    tokens = sorted(tokens, key=lambda x: x[1], reverse=True)
+    # Drop frequency data
+    tokens = [token for token, freq in tokens]
+    # Only include the first N elements
+    tokens = tokens[:top_n]
+    return tokens
 
 
-def create_language_profile(language: str, text: str, stop_words: list) -> dict or None:
+def create_language_profile(language: str,
+                            text: str,
+                            stop_words: list) -> dict or None:
     """
     Creates a language profile
     :param language: a language
     :param text: a text
     :param stop_words: a list of stop words
     :return: a dictionary with three keys – name, freq, n_words
     """
-    pass
+    # Check for bad input
+    if (not isinstance(language, str)
+            or not isinstance(text, str)
+            or not isinstance(stop_words, list)):
+        return None
+    # Get frequencies
+    tokens = tokenize(text)
+    tokens = remove_stop_words(tokens, stop_words)
+    frequencies = calculate_frequencies(tokens)
```

Comment:
or you can join 111-113 in one line if you wish
or at least join 111-112, as you don't really need a list of tokens further